### PR TITLE
Initial coding

### PR DESCRIPTION
**Pull request by brianolson**
Enough structs and handlers to query Account info, Transactions, and submit a basic APT transfer transaction.

TODO: more unit tests, more API wrappers, better structs, etc.

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+type AccountResourceInfo struct {
+	Type string         `json:"type"`
+	Data map[string]any `json:"data"` // TODO: what are these? Build a struct.
+}
+
+func (rc *RestClient) AccountResource(address AccountAddress, resourceType string, ledger_version ...int) (data map[string]any, err error) {
+	au := rc.baseUrl
+	// TODO: offer a list of known-good resourceType string constants
+	au.Path = path.Join(au.Path, "accounts", address.String(), "resource", resourceType)
+	if len(ledger_version) > 0 {
```
Comment (gregnazario) [inline client.go:142]: Personally, prefer descriptive names over short 2 character variables.

File: client.go [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+		err = fmt.Errorf("error getting response data, %w", err)
+		return
+	}
+	response.Body.Close()
+	err = json.Unmarshal(blob, &data)
+	return
+}
+
+// Deprecated-ish, #SubmitTransaction() should be much faster and better in every way
+func (rc *RestClient) TransactionEncode(request map[string]any) (data []byte, err error) {
```
Comment (gregnazario) [inline client.go:301]: Let's kill it then unless we need it.

**Review by gregnazario — APPROVED** [review]
Let's get this in.  Will need to do a full review of everything again afterwards anyways / expect there to be a bunch of style etc. churn

File: account.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	"errors"
+	"fmt"
+	"strings"
+
+	"golang.org/x/crypto/sha3"
+)
+
+type AccountAddress [32]byte
+
+var Account0x1 AccountAddress = AccountAddress{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1}
```
[the remark this answers is not in the document]
Comment (brianolson) [reply account.go:17]: we could move account stuff to a package, e.g.
```
import "github.com/aptos-labs/aptos-go-sdk/acccount"
foo := account.ONE
```

**Pull request merged**